xml-jaxp/src/main/java/org/jboss/as/quickstart/xml/jaxp/dom/DOMXMLParser.java

```
@@ -0,0 +1,160 @@
+            if (childName == null)
+                continue;
+
+            if (n.getLocalName().equals("book")) {
+                Book b = parseBook((Element) n);
+                catalog.add(b);
+            }
+        }
+
+        return catalog;
+
+    }
+
+    private Book parseBook(Element n) {
+        Book b = new Book();
+        NodeList children = n.getChildNodes();
+        for (int index = 0; index < children.getLength(); index++) {
+            Node child = children.item(index);
+            String childName = child.getLocalName();
+            if (childName == null)
+                continue;
+
+            if (childName.equals("author")) {
+                Element childElement = (Element) child;
+
+                String textVal = getTextValue(childElement);
+                b.setAuthor(textVal);
+            } else if (childName.equals("title")) {
+                Element childElement = (Element) child;
+
+                String textVal = getTextValue(childElement);
+                b.setTitle(textVal);
+            } else if (childName.equals("genre")) {
+                Element childElement = (Element) child;
+
+                String textVal = getTextValue(childElement);
+                b.setGenre(textVal);
+            } else if (childName.equals("price")) {
+                Element childElement = (Element) child;
+
+                String textVal = getTextValue(childElement);
+                b.setPrice(Float.parseFloat(textVal));
+            } else if (childName.equals("publish_date")) {
+                Element childElement = (Element) child;
+
+                String textVal = getTextValue(childElement);
+                Date d;
+                try {
+
+                    d = DATE_FORMATTER.parse(textVal);
+                    b.setPublishDate(d);
+                } catch (ParseException e) {
+                    // TODO Auto-generated catch block
+                    e.printStackTrace();
```

Again, maybe explain why we are swallowing this exception, or re-throw it?